python/perf-kernels/flash-attention.py

```diff
@@ -299,35 +290,36 @@ def _attn_fwd_inner(
 
 @triton.jit
 def attn_fwd(
-    Q, K, V, bias, sm_scale, M, Out,
+    Q, K, V, B, sm_scale, M, Out,
     stride_qz, stride_qh, stride_qm, stride_qk,
     stride_kz, stride_kh, stride_kn, stride_kk,
     stride_vz, stride_vh, stride_vk, stride_vn,
     stride_oz, stride_oh, stride_om, stride_on,
     stride_bz, stride_bh, stride_bm, stride_bn,
     cu_seqlens_q, cu_seqlens_k,
+    max_seqlens_q, max_seqlens_k,
+    head_dim_q, head_dim_k,
     dropout_p,
     philox_seed,
     philox_offset_base,
     encoded_softmax,
-    actual_block_dmodel,
-    max_seqlens_q, max_seqlens_k,
     VARLEN: tl.constexpr,
-    hq, hk,
     STAGE: tl.constexpr,
     BLOCK_M: tl.constexpr,
     BLOCK_DMODEL: tl.constexpr,
     BLOCK_N: tl.constexpr,
     PRE_LOAD_V: tl.constexpr,
-    BIAS_TYPE: tl.constexpr,
     ENABLE_DROPOUT: tl.constexpr,
-    RETURN_ENCODED_SOFTMAX: tl.constexpr
+    RETURN_ENCODED_SOFTMAX: tl.constexpr,
+    BIAS_TYPE: tl.constexpr,
+    PADDED_HEAD: tl.constexpr,  # Cannot be inferred by AOT Compiler
 ):
-    is_mqa = hq != hk
+    is_mqa = head_dim_q != head_dim_k
```

Comment:
This is not right. MQA / GQA depends on the number of Q and KV heads being different. It is not possible to have legal attention GEMMs with different head dims.